tests/test_transf_comp.py

```diff
@@ -1,29 +1,33 @@
-import unittest
+import pytest
 import numpy as np
 from cpmpy import *
 from cpmpy.transformations.flatten_model import flatten_constraint
 from cpmpy.transformations.comparison import only_numexpr_equality
 from cpmpy.expressions.variables import _IntVarImpl, _BoolVarImpl # to reset counters
 
-class TestTransfComp(unittest.TestCase):
-    def setUp(self):
-        _IntVarImpl.counter = 0
-        _BoolVarImpl.counter = 0
+def _generate_inputs():
```

Comment:
You here separate the logic of the test from the checking of the test. That doesn't feel right... which transformation you apply determines what the input/output mapping is, so that code should stay together?

Reply:
I agree with this pattern. In this case, the transformation is always the same so I didn't make it part of the parametrization, but that's the idea when you start testing different transforms (like in test_constraints, you have different solvers).